### PR TITLE
changed credit card payment flow

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: Api/Services/Accommodations/Bookings/CreditCardPaymentConfirmationService.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            _administratorContext = administratorContext;
+            _edoContext = edoContext;
+            _documentsService = documentsService;
+            _notificationService = notificationService;
+        }
+
+        public async Task<Result<Booking, ProblemDetails>> Confirm(int bookingId)
+        {
+            return await GetBooking()
+                .Tap(SendReceipt)
```
Comment (mingazhev) [inline Api/Services/Accommodations/Bookings/CreditCardPaymentConfirmationService.cs:30]: We've decided to use tap only for situation when result is not modified.
If you want to use the Result, use Bind

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: HappyTravel.Edo.Data/Booking/CreditCardPaymentConfirmation.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using System;
+
+namespace HappyTravel.Edo.Data.Booking
+{
+    public class CreditCardPaymentConfirmation
+    {
+        public int BookingId { get; set; }
+        public Booking Booking { get; set; }
+        public int AgentId { get; set; }
```
[the remark this answers is not in the document]
Comment (DFSko) [reply HappyTravel.Edo.Data/Booking/CreditCardPaymentConfirmation.cs:9]: Administrator has agency Id?

[the remark this answers is not in the document]
Comment (DFSko) [reply HappyTravel.Edo.Data/Booking/CreditCardPaymentConfirmation.cs:9]: Maybe rename AgentId to AdministratorId?

File: Api/Services/Accommodations/Bookings/CreditCardPaymentConfirmationService.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                var email = await _edoContext.Agents
+                    .Where(a => a.Id == booking.AgentId)
+                    .Select(a => a.Email)
+                    .SingleOrDefaultAsync();
+
+                await _notificationService.SendReceiptToCustomer(receiptInfo, email);
+                return booking;
+            }
+
+            async Task<Result<Booking, ProblemDetails>> SaveConfirmation(Booking booking)
```
[the remark this answers is not in the document]
Comment (DFSko) [reply Api/Services/Accommodations/Bookings/CreditCardPaymentConfirmationService.cs:61]: Fixed

File: Api/Services/Accommodations/Bookings/CreditCardPaymentConfirmationService.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Accommodations/Bookings/CreditCardPaymentConfirmationService.cs]: Fixed

File: HappyTravel.Edo.Data/Booking/CreditCardPaymentConfirmation.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mingazhev) [reply HappyTravel.Edo.Data/Booking/CreditCardPaymentConfirmation.cs:9]: Sure, AdministratorId

**Pull request closed without merging**